src/Microsoft.IdentityModel.Tokens/EventBasedLRUCache.cs

```diff
@@ -331,15 +349,14 @@ private void RemoveItemFromLinkedList(LRUCacheItem<TKey, TValue> newCacheItem)
         private void EventQueueTaskAction()
         {
             Interlocked.Increment(ref _taskCount);
-
-            // Keep running until it is disposed, or expired (DateTime.UtcNow > _eventQueueTaskEndTime).
-            while (!_disposed)
+            try
```

Comment:
The try/catch should be moved to inside the while loop so if any of the actions throws exception it will continue.

Reply:
Fixed!